### PR TITLE
Add Repeat and Retry functionality to sigh test

### PR DESCRIPTION
**Pull request by thebrianchen**
Adding the option to allow retries and repeat tests. I initially tried adding repeat functionality directly into mocha by encapsulating `runner.run()` which would allow test pass/fail/end hooks to be added more easily. However, I ran into a mocha implementation [issue](https://github.com/mochajs/mocha/issues/2706) that requires the entire mocha instance to be re-instantiated and the 'require' cache cleared in between runs. 

I removed the return from `test()` -- not sure where it's being used.

Also, not sure if the package.lock should be excluded // how to auto-exclude it from future commits.

### CODE REVIEW

**Comment by thebrianchen** [comment]
I signed it!

File: tools/sigh.js [hunk 1]
```diff
@@ -581,20 +589,23 @@ function test(args) {
...
-        '--no-deprecation',
-        ...extraFlags,
-        '--loader',
-        fixPathForWindows(path.join(__dirname, 'custom-loader.mjs')),
-        '-r',
-        'source-map-support/register.js',
-        runner
-      ],
-      {stdio: 'inherit'});
+  for (var i = 0; i < JSON.stringify(options.repeat || 1); i++) {
```
Comment (lindner) [inline tools/sigh.js]: Needs some simple docs.  A comment will do.

Can you hoist the options.repeat value into a constant.  That will make it obvious what's happening,

File: tools/sigh.js [hunk 2]
```diff
@@ -581,20 +589,23 @@ function test(args) {
...
+    console.log('RUN %s:', i+1);
```
Comment (lindner) [inline tools/sigh.js]: Optional: consider outputting a timestamp.

File: devtools/package-lock.json [hunk 3]
```diff
@@ -52,11 +52,6 @@
         "@polymer/polymer": "^3.0.0"
```
Comment (lindner) [inline devtools/package-lock.json]: omit this file from your PR.  Thanks!

**Comment by lindner** [comment]
Check the return-code you're generating?  If any test fails you should return non-zero for the test phase. 

```
😱
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! arcs@0.0.0 test-sigh: `cross-env tools/sigh`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the arcs@0.0.0 test-sigh script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2019-01-15T21_18_55_679Z-debug.log
```

File: tools/sigh.js [hunk 4]
```diff
@@ -536,6 +538,12 @@ function test(args) {
         mocha.suite.emit('require', null, '${test}', mocha);
         mocha.suite.emit('post-require', global, '${test}', mocha);
       `);
+      if (options.retries) {
```
Comment (lindner) [inline tools/sigh.js]: validate value?

also minimist docs say:

""Numeric-looking arguments will be returned as numbers unless opts.string or opts.boolean is set for that argument name."""

Comment (thebrianchen) [reply tools/sigh.js]: I added validation just now to ensure that is a number. I tried adding a `console.log()` for non-numeric inputs, but logging it in `buildTestRunner()` would print it multiple times.

**Comment by lindner** [comment]
LGTM, Raul for approval.

**Comment by raulverag** [comment]
I don't see why we would want to repeat or retry tests within one run. What does it even mean if the tests pass on one repeat and then fail on another, other than that the tests are flaky? Furthermore, it seems that if the last run succeeds, the user won't even see any failing output, because that will have scrolled off the screen. Correct me if I'm mistaken, and please elaborate as to what purpose repeats and retries serve.

**Comment by shans** [comment]
Friendly reminder to please discuss changes (such as this) which have the potential to change development practices for all of us in a team meeting, ideally before starting development.

**Comment by lindner** [comment]
For folks tuning in, this work was done to specifically address issue  #779

My apologies for not bringing this up in a larger context.  It was deemed something simple enough as a good-first-issue.

**Comment by lindner** [comment]
Raul suggested that this needs to have summary output at the end.

**Comment by raulverag** [comment]
Yes. If the idea is to help diagnose flaky tests, as described in #779, then the results must be displayed in some kind of summary, e.g. "x out of y tests succeeded. First failure was repeat N.", or in the case of retries, "Tests succeeded after X retries".

**Comment by thebrianchen** [comment]
The mocha.js implementation of `retries` does not allow for logging and will simply run the test until it either passes or the retry count is exhausted, so I removed the the `retries` flag option.

Added a summary to the end of the runs. Example output:
```
100 runs completed. 4 runs failed.
Failed runs: [4, 47, 49, 81]
```

**Comment by lindner** [comment]
LGTM, Raul for final approvals.

**Pull request merged**